docs/development/visualize/development-create-visualization.asciidoc

```diff
@@ -0,0 +1,307 @@
+-----------
+import { VisFactoryProvider } from 'ui/vis/vis_factory';
+
+class MyVisualization {
+   constructor(vis, el) {
+      this.el = el;
+      this.vis = vis;
+   }
+   render(visData, status) {
+      return new Promise(resolve => {
+         ...
+         resolve('done rendering');
+      }),
+   destroy() {
+      console.log('destroying');
+   }
+}
+
+export default function MyNewVisType(Private) {
+  const VisFactory = Private(VisFactoryProvider);
+
+  return VisFactory.createBaseVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    visualization: MyVisualization
+  });
+}
+-----------
+
+[[development-angular-visualization-type]]
+==== Angular Visualization Type
+Angular visualization type assumes you are using angular as your rendering technology. Instead of providing the
+controller we need to provide the angular template to render. 
+
+Visualization will receive vis, uiState and visData on the $scope and needs to call $scope.renderComplete() once 
+its done rendering.
+
+["source","js"]
+-----------
+export default function MyNewVisType(Private) {
+  const VisFactory = Private(VisFactoryProvider);
+  
+  return VisFactory.createAngularVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    visConfig: {
+       template: '<div ng-controller="MyAngularController"></div>`
+    }
+  });
+}
+-----------
+
+[[development-react-visualization-type]]
+==== React Visualization Type
+React visualization type assumes you are using React as your rendering technology. Instead of passing it an angular
+template you need to pass React component.
+
+Visualization will receive vis, uiState and visData as props as well as the renderComplete which should be called once
+done rendering.
+
+["source","js"]
+-----------
+import { ReactComponent } from './my_react_component';
+
+export default function MyNewVisType(Private) {
+  const VisFactory = Private(VisFactoryProvider);
+  
+  return VisFactory.createReactVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    visConfig: {
+       template: ReactComponent
+    }
+  });
+}
+-----------
+
+[[development-vislib-visualization-type]]
+==== Vislib Visualization Type
+This visualization type should only be used for `vislib` visualizations. Vislib is kibana's D3 library which can produce
+point series charts and pie charts.
+
+[[development-vis-editors]]
+=== Visualization Editors
+By default visualization will use the `default` editor (which is the sidebar editor you see in current kibana charts) but can be changed. Currently no other editors are provided (`timelion` editor will probably be added in the near future). Plugin can register a new editor in registry to make it available to other visualizations, but is not necessary (if its a one-time thing which will not be reused)
```

Comment:
actually timelion editor is already added its just not exposed to other visualizations at this time. removing this part of docs.